src/CodeStyle/Core/CodeFixes/Host/Mef/CodeStyleHostLanguageServices.MefHostExportProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Composition.Hosting;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal sealed partial class CodeStyleHostLanguageServices : HostLanguageServices
+    {
+        private sealed class MefHostExportProvider : IMefHostExportProvider
+        {
+            private readonly CompositionHost _compositionContext;
+
+            public MefHostExportProvider(string languageName)
+            {
+                var assemblies = CreateAssemblies(languageName);
+                var compositionConfiguration = new ContainerConfiguration().WithAssemblies(assemblies);
+                _compositionContext = compositionConfiguration.CreateContainer();
+            }
+
+            private static ImmutableArray<Assembly> CreateAssemblies(string languageName)
+            {
+                using var disposer = ArrayBuilder<string>.GetInstance(out var assemblyNames);
+
+                assemblyNames.Add("Microsoft.CodeAnalysis.CodeStyle.Fixes");
+                switch (languageName)
+                {
+                    case LanguageNames.CSharp:
+                        assemblyNames.Add("Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes");
+                        break;
+
+                    case LanguageNames.VisualBasic:
+                        assemblyNames.Add("Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes");
+                        break;
+                }
+
+                return MefHostServices.DefaultAssemblies.Concat(
+                    MefHostServicesHelpers.LoadNearbyAssemblies(assemblyNames));
+            }
+
+
+            IEnumerable<Lazy<TExtension>> IMefHostExportProvider.GetExports<TExtension>()
+            {
+                return _compositionContext.GetExports<TExtension>().Select(e => new Lazy<TExtension>(() => e));
+            }
+
+            IEnumerable<Lazy<TExtension, TMetadata>> IMefHostExportProvider.GetExports<TExtension, TMetadata>()
+            {
+                var importer = new WithMetadataImporter<TExtension, TMetadata>();
+                _compositionContext.SatisfyImports(importer);
+                return importer.Exports;
+            }
+
+            private class WithMetadataImporter<TExtension, TMetadata>
+            {
+                [ImportMany]
+                public IEnumerable<Lazy<TExtension, TMetadata>> Exports { get; set; }
+            }
+        }
+    }
+}
```

Comment:
skipping this file.  i will leave to mef people to look at :)

Reply:
Yes, this is most likely a temporary approach for discovering language services in CodeStyle layer. https://github.com/dotnet/roslyn/issues/41462 tracks removal of MEF, and likely would need @jasonmalinowski or @sharwell to figure out a good non-MEF based solution to implement it.